lms/djangoapps/django_comment_client/base/event_transformers.py

```diff
@@ -0,0 +1,154 @@
+"""
+Transformers for Discussion-related events.
+"""
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse, NoReverseMatch
+
+from eventtracking.processors.exceptions import EventEmissionExit
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locator import CourseLocator
+
+from django_comment_client.base.views import add_truncated_title_to_event_data
+from django_comment_client.permissions import get_team
+from django_comment_client.utils import get_cached_discussion_id_map_by_course_id
+from track.transformers import EventTransformer, EventTransformerRegistry
+
+
+def _get_string(dictionary, key, del_if_bad=True):
+    """
+    Get a string from a dictionary by key.
+
+    If the key is not in the dictionary or does not refer to a string:
+        - Return None
+        - Optionally delete the key (del_if_bad)
+    """
+    if key in dictionary:
+        value = dictionary[key]
+        if isinstance(value, basestring):
+            return value
+        else:
+            if del_if_bad:
+                del dictionary[key]
+            return None
+    else:
+        return None
+
+
+@EventTransformerRegistry.register
+class ForumThreadViewedEventTransformer(EventTransformer):
+    """
+    Transformer for forum-thread-viewed mobile navigation events.
+    """
+
+    match_key = u'edx.bi.app.navigation.screen'
+
+    def process_event(self):
+        """
+        Process incoming mobile navigation events.
+
+        For forum-thread-viewed events, change their names to
+        edx.forum.thread.viewed and manipulate their data to conform with
+        edx.forum.thread.viewed event design.
+
+        Throw out other events.
+        """
+
+        # Get event context dict
+        # Throw out event if context nonexistent or wrong type
+        context = self.get('context')
+        if not isinstance(context, dict):
+            raise EventEmissionExit()
+
+        # Throw out event if it's not a forum thread view
+        if _get_string(context, 'label', del_if_bad=False) != 'Forum: View Thread':
```

Comment:
Where is the string  'Forum: View Thread' defined in the tracking code? I only see you emitting it in the tests.